pkg/rancher-desktop/backend/backendHelper.ts

```diff
@@ -116,4 +121,68 @@ export default class BackendHelper {
   static requiresCRIDockerd(engineName: string, kubeVersion: string | semver.SemVer): boolean {
     return engineName === ContainerEngine.MOBY && semver.gte(kubeVersion, '1.24.1') && semver.lte(kubeVersion, '1.24.3');
   }
+
+  /**
+   * Validate the cfg.kubernetes.version string
+   * If it's valid and available, use it.
+   * Otherwise fall back to the first (recommended) available version.
+   */
+  static async getDesiredVersion(currentConfigVersionString: string|undefined, availableVersions: semver.SemVer[], noModalDialogs: boolean, settingsWriter: (_: any) => void): Promise<semver.SemVer> {
+    let storedVersion: semver.SemVer|null;
+    let matchedVersion: semver.SemVer|undefined;
+    const invalidK8sVersionMainMessage = `Requested kubernetes version '${ currentConfigVersionString }' is not a valid version.`;
+
+    if (currentConfigVersionString) {
+      storedVersion = semver.parse(currentConfigVersionString);
+      if (storedVersion) {
+        matchedVersion = availableVersions.find((v) => {
+          try {
+            return v.compare(storedVersion as semver.SemVer) === 0;
+          } catch (err: any) {
+            console.error(`Can't compare versions ${ storedVersion } and ${ v }: `, err);
+            if (!(err instanceof TypeError)) {
+              return false;
+            }
+            // We haven't seen a non-TypeError exception here, but it would be worthwhile to have it reported.
+            // This throw will cause the exception to appear in a non-fatal error reporting dialog box.
+            throw err;
+          }
+        });
+        if (matchedVersion) {
+          return matchedVersion;
+        }
+      }
+    }
+    // If we're here either there's no existing cfg.k8s.version, or it isn't valid
+    if (!availableVersions.length) {
+      if (currentConfigVersionString) {
+        console.log(invalidK8sVersionMainMessage);
+      } else {
+        console.log('Internal error: no available kubernetes versions found.');
+      }
+      throw new Error('No kubernetes version available.');
+    }
```

Comment:
Do we want to move the check for `availableVersions` to the beginning of this function? It looks to me like we can't do anything without a collection of `avaialableVersions` so we'd be able to handle the error case without having to go through the above block first.

Reply:
good point. Let's me break two if blocks into one.